app/models/user.rb

```diff
@@ -234,13 +234,15 @@ def self.try_to_login(login, password, api_request = false)
       # user has an authentication method and the authentication was successful
       if api_request && user.authenticate_by_personal_access_token(password)
         logger.debug("Authenticated user #{user.login} with a Personal Access Token.")
+        user.post_successful_login
```

Comment:
this should reflect the unless below, probably convert to if ... else $original code end